base/applications/mscutils/servman/progress.c

```diff
@@ -76,13 +78,20 @@ unsigned int __stdcall ActionThread(void* Param)
                             IDS_PROGRESS_INFO_START);
 
         /* Start the service */
-        if (DoStartService(ProgressData->ServiceName,
+        dwResult = DoStartService(ProgressData->ServiceName,
                            ProgressData->hProgress,
-                           ProgressData->Param))
+                           ProgressData->Param);
+        if (dwResult==SC_MANAGER_SUCCESS)
         {
             /* We're done, slide the progress bar up to the top */
             CompleteProgressBar(ProgressData->hProgress);
         }
+        else
+        {
+            LoadStringW(hInstance, IDS_ERROROCCURED, szMessage, _countof(szMessage));
```

Comment:
What about creating a helper function "DisplayError(DWORD dwError)" whose purpose is to do exactly all that, instead of copy-pasting the same lines of codes 5 times in the code?

Also, why not displaying instead the messages corresponding to the standard returned last errors instead?

Reply:
Good idea !